### ISSUE TITLE
Loop on array of objects

### ISSUE THREAD

**Issue by scerelli** [kaisermann/svelte-i18n#83]
**Describe the bug**
Loop on array of objects

**To Reproduce**
```
team: {
 teammemberName.career: [
  {
    company: "Company name",
    description: "Whatevah here",
  }
]
```

then try to do something like this:
```
{#each $_(`team.${selectedMember}.career`) as c, i}
 <h5>{$_(`team.${selectedMember}.career.${i}.company`)}</h5>
{/each}
```
of course this won't work ```#each $_(`team.${selectedMember}.career`)``` loops more times than the actual length of career. 

by reading the tests i discovered that the lib is flattening the array so it means that this each loop is looping on the strings instead of the array. Is this an expected behavior? what's the correct way to do this stuff?

**Comment by kaisermann** [comment]
Hey @scerelli 👋 

This is expected behavior, as you can see in #50. The reason behind flattening the dictionaries is to simplify the merging of multiple partial dictionaries. 

Also, I don't have experience with a lot of i18n libraries, do you know any that allow this kind of behavior?

**Comment by MichielLeyman** [comment]
I've written a custom `getArrayFromLocale` workaround to make this possible. 

> Also, I don't have experience with a lot of i18n libraries, do you know any that allow this kind of behavior?

`i18next` does this: https://www.i18next.com/translation-function/objects-and-arrays

I don't have much experience with i18n, but iterating over logical groups makes a lot of sense and it's the first thing I looked for in the library. I implemented `svelte-i18n` after I was already iterating over arrays of objects, not doing so feels like regression.

**Comment by Jerome1337** [comment]
I tried to do the same but we cannot loop over these JSON arrays, here is the workaround I did:

```json
"description": {
  "0": "...",
  "1": "...",
  "2": "...",
  "3": "..."
}
```

```js
{#each new Array(4) as _, index}
  <p>{$t(`description.${index}`)}</p>
{/each}
```

This is working but it would be nice to be able to loop over JSON array

**Comment by kaisermann** [comment]
Fixed in `3.2.0`.

**Comment by blynx** [comment]
This seems to have broken my translations - I like flat data (🤷‍♂️) and have a dictionairy like so:

```
{
    "something": "Something Works Fine",
    "something.broken": "Something Does Not Work Fine"
}
```

The former gets translated, the latter not anymore with the "not found" warning `[svelte-i18n] The message "my.message" was not found in "de", "en".`  
I wonder if this is intended or if there could/should be a fallback to retrieve the dot separated keys? This is not a major issue for me since I can just switch to another separator - but I was wondering you are aware of this?

cheers, and thanks for the library! :)

**Comment by kaisermann** [comment]
Hey @blynx 👋 Sorry, my bad! Fixed in `3.2.5` 🎉

**Comment by blynx** [comment]
wow, that was quick! Thank you! 🙌
